### PR TITLE
Feature Request: Add changepoint_range resolve #217

### LINKED ISSUE #217: Feature Request: Expose Potential Changepoint 80% Threshold?
I think it would make sense to expose a class variable to allow the user to change the 80% threshold for potential automatic changepoints. For long time series, the 20% runway may be longer than is necessary. 

Thanks,
Scott

### PR DESCRIPTION
**Pull request by MIsmailKhan**
Provide necessary changes to pull request "Adding changepoint threshold #299"

* Rename changepoint_threshold to changepoint_range 
* Modified changepoint_range Docstring 
* Make changepoint_ range validation in validate_inputs function.

### CODE REVIEW

**Comment by MIsmailKhan** [comment]
There is some doubt on the interval for changepoint_range. @bletham has stated it as [0,1] while @zippeurfou's pull request has defined it as (0,1].

Suggestion: In the changepoint_range validation, perhaps it would be better if a warning is thrown instead of an error followed by a re-input from user.

File: R/R/prophet.R [hunk 1]
```diff
@@ -443,9 +447,14 @@ set_changepoints <- function(m) {
...
     }
   } else {
-    # Place potential changepoints evenly through the first 80 pcnt of
+    # Place potential changepoints evenly through the first changepoint.range proportion of
     # the history.
-    hist.size <- floor(nrow(m$history) * .8)
+    if (m$changepoint.range > 1 || m$changepoint.range <= 0){
+      m$changepoint.range <- .8
+      message('changepoint.range greater than 1 or less than equal to 0. Using ',m$changepoint.range)
+    }
```
Comment (bletham) [inline R/R/prophet.R]: This block should be replaced by a check inside `validate_inputs` that raises an error if not valid, like in the Python version in this commit.

**Comment by bletham** [comment]
Sorry for the delay on this review too. I'll merge this in asap once these remaining things are resolved.

**Comment by bletham** [comment]
I went ahead and merged #299 and made the necessary updates in https://github.com/facebook/prophet/commit/cb0b47994b1d2d6612f75eff4b9d4bec6f5ad846 so this feature is good-to-go now.

**Pull request closed without merging**